llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -696,7 +734,7 @@ void VPInstruction::execute(VPTransformState &State) {
     assert(GeneratedValue && "generatePerPart must produce a value");
     assert((GeneratedValue->getType()->isVectorTy() ==
                 !GeneratesPerFirstLaneOnly ||
-            State.VF.isScalar()) &&
+            State.VF.isSingleScalar()) &&
```

Comment:
Checking if VF is 1 should stay isScalar()?

Reply:
Ah yes, should be fixed, thanks!